test/unit/RelayServer.test.ts

```diff
@@ -427,4 +440,102 @@ describe('RelayServer', () => {
+                    },
+                    mockDependencies
+                );
+            });
+
+            afterEach(function () {
+                restore();
+            });
+
+            it('shold not sponsor relay transaction', function () {
+                const relayRequest: RelayRequest = {
+                    request: {
+                        to: ''
+                    }
+                } as RelayRequest;
+                expect(server.isSponsoredTx(relayRequest), 'Tx is sponsored').to
+                    .be.false;
+            });
+
+            it('shold not sponsor deploy transaction', function () {
+                const deployRequest: DeployRequest = {
+                    request: {
+                        to: ''
+                    }
+                } as DeployRequest;
+                expect(server.isSponsoredTx(deployRequest), 'Tx is sponsored')
+                    .to.be.false;
+            });
+
+            it('should sponsor relay transaction if its sponsored destination', function () {
+                const relayRequest: RelayRequest = {
+                    request: {
+                        to: '0x1'
+                    }
+                } as RelayRequest;
+                expect(
+                    server.isSponsoredTx(relayRequest),
+                    'Tx is not sponsored'
+                ).to.be.true;
+            });
+
+            it('should sponsor deploy transaction if its sponsored destination', function () {
+                const deployRequest: DeployRequest = {
+                    request: {
+                        to: '0x1'
+                    }
+                } as DeployRequest;
+                expect(
+                    server.isSponsoredTx(deployRequest),
+                    'Tx is not sponsored'
+                ).to.be.true;
+            });
+        });
+
+        describe('disabledSponsoredTx(false)', function () {
+            let server: RelayServer;
+            const relayRequest: RelayRequest = {
+                request: {
+                    to: ''
+                }
+            } as RelayRequest;
+            const deployRequest: DeployRequest = {
+                request: {
+                    to: ''
+                }
+            } as DeployRequest;
+
+            beforeEach(function () {
+                server = new RelayServer(
+                    { disableSponsoredTx: false },
+                    mockDependencies
+                );
+            });
+
+            it('should sponsor relay transaction', function () {
+                expect(
+                    server.isSponsoredTx(relayRequest),
+                    'Tx is not sponsored'
+                ).to.be.true;
+            });
+
+            it('should sponsor deploy transaction', function () {
+                expect(
+                    server.isSponsoredTx(deployRequest),
+                    'Tx is not sponsored'
+                ).to.be.true;
+            });
+        });
+    });
```

Comment:
I'd also include some cases:
 - `sponsoredDestinations` is empty
 - `sponsoredDestinations` includes more than one address.

Reply:
updated